src/installer/tests/Microsoft.NET.HostModel.Tests/Microsoft.NET.HostModel.AppHost.Tests/AppHostUsedWithSymbolicLinks.cs

```diff
@@ -0,0 +1,234 @@
+        }
+
+        [Fact]
+        public void Put_satellite_assembly_behind_symlink()
+        {
+            // Creating symbolic links requires administrative privilege on Windows, so skip test.
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                return;
+
+            var fixture = sharedTestState.StandaloneAppFixture_Localized
+                .Copy();
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // Set code page to output unicode characters.
+                Command.Create("chcp 65001").Execute();
+            }
+
+            var appExe = fixture.TestProject.AppExe;
+            var binDir = fixture.TestProject.OutputDirectory;
+            var satellitesDir = Path.Combine(Directory.GetParent(fixture.TestProject.Location).ToString(), "PutSatellitesSomewhereElse");
+            Directory.CreateDirectory(satellitesDir);
+
+            var firstSatelliteDir = Directory.GetDirectories(binDir).Single(dir => dir.Contains("kn-IN"));
+            var firstSatelliteNewDir = Path.Combine(satellitesDir, "kn-IN");
+            Directory.Move(firstSatelliteDir, firstSatelliteNewDir);
+            CreateSymbolicLink(firstSatelliteDir, firstSatelliteNewDir);
+
+            var secondSatelliteDir = Directory.GetDirectories(binDir).Single(dir => dir.Contains("ta-IN"));
+            var secondSatelliteNewDir = Path.Combine(satellitesDir, "ta-IN");
+            Directory.Move(secondSatelliteDir, secondSatelliteNewDir);
+            CreateSymbolicLink(secondSatelliteDir, secondSatelliteNewDir);
+
+            Command.Create(appExe)
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Execute()
+                .Should().Pass()
+                .And.HaveStdOutContaining("ನಮಸ್ಕಾರ! வணக்கம்! Hello!");
+        }
+
+        public class SharedTestState : IDisposable
+        {
+            public TestProjectFixture StandaloneAppFixture_Localized { get; }
+            public TestProjectFixture StandaloneAppFixture_Published { get; }
+            public RepoDirectoriesProvider RepoDirectories { get; }
+
+            public SharedTestState()
+            {
+                RepoDirectories = new RepoDirectoriesProvider();
+
+                var localizedFixture = new TestProjectFixture("LocalizedApp", RepoDirectories);
+                localizedFixture
+                    .EnsureRestoredForRid(localizedFixture.CurrentRid)
+                    .PublishProject(runtime: localizedFixture.CurrentRid);
+
+                var publishFixture = new TestProjectFixture("StandaloneApp", RepoDirectories);
+                publishFixture
+                    .EnsureRestoredForRid(publishFixture.CurrentRid)
+                    .PublishProject(runtime: publishFixture.CurrentRid);
+
+                StandaloneAppFixture_Localized = localizedFixture;
+                StandaloneAppFixture_Published = publishFixture;
```

Comment:
Couple more tests we might want to add:
* Framework dependent app - run it from a symlink
* Framework dependent app where runtime install location is a symlink (basically `DOTNET_ROOT` points to a symlink)
* Running an app via `dotnet app.dll` where the `dotnet` is a symlink. (I think this is close to the snap scenario)

Reply:
Some more
* Make sure we have a test to validate additional probing paths (that they continue to work even after this change)